File: src/main/java/org/dxworks/githubminer/utils/TestUtils.java
package org.dxworks.githubminer.utils;

import lombok.SneakyThrows;
import org.dxworks.utils.java.rest.client.providers.BasicAuthenticationProvider;

import java.util.Properties;

public class TestUtils {

    private static Properties properties;

    public static String getProperty(String key) {
        if (properties == null)
            loadProperties();

        return properties.getProperty(key);
    }

    @SneakyThrows
    private static void loadProperties() {
        properties = new Properties();
        properties.load(TestUtils.class.getClassLoader().getResourceAsStream("test.properties"));
    }

    public static BasicAuthenticationProvider getGithubCredentials() {
        return new BasicAuthenticationProvider(getProperty("test.github.username"), getProperty("test.github.token"));
    }

    public static BasicAuthenticationProvider getJiraCredentials() {
        return new BasicAuthenticationProvider(getProperty("test.jira.username"), getProperty("test.jira.token"));
    }
}

File: src/main/kotlin/org/dxworks/githubminer/utils/TestUtils.kt
package org.dxworks.githubminer.utils

import org.dxworks.githubminer.constants.ANONYMOUS
import java.io.FileInputStream
import java.nio.file.Paths
import java.util.*

object TestUtils {
    private var properties: Properties? = null
    fun getProperty(key: String?): String {
        if (properties == null) loadProperties()
        return properties!!.getProperty(key)
    }

    private fun loadProperties() {
        properties = Properties()
        properties!!.load(FileInputStream(Paths.get("config", "test.properties").toFile()))
    }

    val githubCredentials: List<String>
        get() = properties?.entries?.filter { (it.key as String).startsWith("test.github.token") }?.map { it.value as String }
                ?: listOf(ANONYMOUS)

}
